packages/flutter/lib/src/widgets/_web_browser_detection_io.dart

```diff
@@ -0,0 +1,12 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// The widget in this file is an empty mock for non-web platforms. See
+// `_web_browser_detection.dart` for the web implementation.
+
+/// A class that provides information about the current browser when on the web.
+class WebBrowserDetection {
```

Comment:
There might be a better home for this rather than in `widgets`

Reply:
What's the main purpose of creating this abstraction?

Reply:
> There might be a better home for this rather than in `widgets`.

Maybe `services`?

Reply:
Another option might be `foundation`, that is where `kIsWeb` lives and `defaultTargetPlatform`.

Reply:
> What's the main purpose of creating this abstraction?

The main purpose for this class is because we can't directly import only `dart:ui_web` on web without a fallback.

Reply:
@mdebbar do you have any thoughts on where this class should live?

Reply:
I ended up just making it private for now, removing the public `web_browser_detection.dart`. The conditional import is now in `editable_text.dart`.